### PR TITLE
Feature multi-model

### PR DESCRIPTION
**Pull request by IntegralProgrammer**
Extends the API so that:
  - A trained model can be selected for an annotation request (eg. HPO, MONDO, etc...)
  - Trained models can be listed (`/lsmodels`)
  - Trained models can be deleted (`/delete_model`)
  - A script can be generated for model training on HPF (`/submit_training_job`)

Modifications were made to the model training code so that it executes on HPF.

### CODE REVIEW

File: docker_run_webapp.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+mkdir ~/ncr_obo_uploads || { echo "~/ncr_obo_uploads ALREADY EXISTS"; exit -1; }
+mkdir ~/ncr_qsub || { echo "~/ncr_qsub ALREADY EXISTS"; exit -1; }
+
+echo "Starting Docker..."
+docker run --rm \
+	-v $(realpath ~/ncr_obo_uploads):/root/uploaded_obo:rw \
+	-v $(realpath ~/ncr_qsub):/root/qsub:rw \
+	-v $(realpath ~/ncr_model_params):/root/opt/ncr/model_params:ro \
+	-p 127.0.0.1:5000:5000 \
+	-e AUTOTEST=$AUTOTEST \
```
Comment (sdumitriu) [inline docker_run_webapp.sh]: This belongs to a different PR.

File: docker_run_webapp.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	-e TEST_IGNORE_SCORE=$TEST_IGNORE_SCORE \
```
Comment (sdumitriu) [inline docker_run_webapp.sh]: This too.

File: app.py [hunk 3]
```diff
@@ -1,20 +1,47 @@
...
+'''
 model = ncrmodel.NCR.loadfromfile('model_params', 'model_params/pmc_model_new.bin')
 threshold = 0.6
-#'''
+'''
+
+#Stored in a form of {"object": model object, "threshold": threshold value}
+NCR_MODELS = {}
 
+#For now, to be later fixed...
```
Comment (sdumitriu) [inline app.py]: When later?

File: app.py [hunk 4]
```diff
@@ -49,6 +76,68 @@ def dated_url_for(endpoint, **values):
     return url_for(endpoint, **values)
 
 
+@app.route('/lsmodels/')
```
Comment (sdumitriu) [inline app.py]: This isn't very RESTful, where paths should be the name of resources, and the HTTP commands should be actions.

So, ideally `GET /models` should list the existing models, `POST /models` (or maybe `PUT /models/HPO`?) should submit a training job, and `DELETE /models/HPO` should delete one model. Can Flask do that easily, route different HTTP methods to different Python methods?

File: app.py [hunk 5]
```diff
@@ -284,12 +377,16 @@ def annotate_post():
 """
 @app.route('/annotate/', methods=['GET'])
 def annotate_get():
+    if not 'model' in request.args:
+        abort(400)
```
Comment (sdumitriu) [inline app.py:354]: I would add some backwards compatibility for this: if there's only one model trained, and there's no `model` specified in the request, then just use the one and only model available.

**Review by sdumitriu — CHANGES_REQUESTED** [review]
I would include only the changes needed for multiple models in this PR, moving the HPF integration in a different PR.

File: docker_run_webapp.sh (hunk 1 above)
Comment (IntegralProgrammer) [reply docker_run_webapp.sh]: Done - this content has been removed and will be later added in a different PR :heavy_check_mark:

File: docker_run_webapp.sh (hunk 2 above)
Comment (IntegralProgrammer) [reply docker_run_webapp.sh]: Done - this content has been removed and will be later added in a different PR :heavy_check_mark:

File: app.py (hunk 3 above)
Comment (IntegralProgrammer) [reply app.py]: I revised this comment clarifying its meaning - see line 18 in `app.py`. In this early version of the multi-model API, models are to be added by modification of the lines below defining the object `NCR_MODELS`. In upcoming Pull Requests, API endpoints will be created for model training, and after successful model training, these models will be automatically added to the `NCR_MODELS` object.

File: app.py (hunk 4 above)
Comment (IntegralProgrammer) [reply app.py]: I have made revisions, you can now `GET /models/` to see the list of available trained models and you can also `DELETE /models/NameOfModelToDelete` to remove a model. Routing different HTTP methods to different Python methods in Flask is simple. `PUT` / `POST` functionality will be added in the PR containing the model training code.

File: app.py (hunk 5 above)
Comment (IntegralProgrammer) [reply app.py:354]: Done - I have added the backwards compatibility as you have described. :heavy_check_mark:

**Comment by IntegralProgrammer** [comment]
Thank you for your feedback. I have revised this PR so that it only includes the changes needed for using multiple models and does not include any model training or HPF code.

**Review by sdumitriu — APPROVED** [review]
